cmd/ocm-backplane/managedJob/createManagedJob.go

```diff
@@ -22,112 +36,182 @@ func newCreateManagedJobCmd() *cobra.Command {
 	}
-	// It is always 1 in length, enforced by cobra
-	canonicalNameArg := args[0]
 
+	// init job cononical name
+	canonicalNameArg := argv[0]
+	options.canonicalName = canonicalNameArg
+
+	// init Cluster key
+	options.clusterId, err = cmd.Flags().GetString("cluster-id")
 	if err != nil {
 		return err
 	}
 
-	client, err := utils.DefaultClientUtils.MakeRawBackplaneAPIClient(backplaneHost)
+	// init URL flag
+	options.url, err = cmd.Flags().GetString("url")
+	if err != nil {
+		return err
+	}
+
+	// init raw flag
+	options.raw, err = cmd.Flags().GetBool("raw")
 	if err != nil {
 		return err
 	}
 
-	cj := BackplaneApi.CreateJobJSONRequestBody{
-		CanonicalName: &canonicalNameArg,
+	return nil
+}
+
+// createJob initiates the job creation in a specific cluster and returns the job info
+func createJob(client BackplaneApi.ClientInterface) (*BackplaneApi.Job, error) {
+
+	jobParams, err := utils.ParseParamsFlag(options.params)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// create job request
+	createJob := BackplaneApi.CreateJobJSONRequestBody{
+		CanonicalName: &options.canonicalName,
 		Parameters: &BackplaneApi.CreateJob_Parameters{
-			AdditionalProperties: parsedParams,
+			AdditionalProperties: jobParams,
 		},
 	}
 
-	// ======== Call Endpoint ========
-	resp, err := client.CreateJob(context.TODO(), clusterID, cj)
+	// call create end point
+	resp, err := client.CreateJob(context.TODO(), options.clusterId, createJob)
 
-	// ======== Render Results ========
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		return utils.TryPrintAPIError(resp, rawFlag)
+		return nil, utils.TryPrintAPIError(resp, options.raw)
 	}
 
+	// format create job response
 	createResp, err := BackplaneApi.ParseCreateJobResponse(resp)
 
 	if err != nil {
-		return fmt.Errorf("unable to parse response body from backplane: \n Status Code: %d", resp.StatusCode)
+		return nil, fmt.Errorf("unable to parse response body from backplane: \n Status Code: %d", resp.StatusCode)
 	}
 
+	// render job details
 	fmt.Printf("%s\nJobId: %s\n", *createResp.JSON200.Message, *createResp.JSON200.JobId)
-	if rawFlag {
+	if options.raw {
 		_ = utils.RenderJsonBytes(createResp.JSON200)
 	}
-	return nil
+	return createResp.JSON200, nil
 }
 
-func init() {
+// waitForCreateJob wait until job status to be Succeeded
+// waitForCreateJob timeouts after 10 min
+func waitForCreateJob(client BackplaneApi.ClientInterface, job *BackplaneApi.Job) error {
+
+	pollErr := wait.PollImmediate(10*time.Second, time.Duration(600)*time.Second, func() (bool, error) {
+		fmt.Printf(".")
+
+		// ========= Get the current job ============
+		jobResp, err := client.GetRun(context.TODO(), options.clusterId, *job.JobId)
+
+		if err != nil {
+			return false, err
+		}
+
+		formatJobResp, err := BackplaneApi.ParseGetRunResponse(jobResp)
+
+		if err != nil {
+			return false, err
+		}
+
+		if *formatJobResp.JSON200.JobStatus.Status == "Succeeded" {
```

Comment:
We should exit and print if the job is failed as well?

Reply:
fixed. 
https://github.com/openshift/backplane-cli/blob/d30d217556c51de11f190bdb0bd4662d29e52a0e/cmd/ocm-backplane/managedJob/createManagedJob.go#L213